src/Cache/Cache.php

```diff
@@ -43,7 +43,7 @@ public function has(string $file): bool
         return \array_key_exists($file, $this->hashes);
     }
 
-    public function get(string $file): ?int
+    public function get(string $file): ?string
```

Comment:
after calling `fromJson` with cache file created before your PR, this method will try to return int, not string, due to content in json file

Reply:
Good catch.

The `signature` will not match as it contains also the PHP CS Fixer `version` in:

https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v3.10.0/src/Cache/FileCacheManager.php#L113

Reply:
before we call `!$this->signature->equals($cache->getSignature())) {`, we do call
`$cache = $this->handler->read();`
`handler->read()` calls `cache = Cache::fromJson($content);`
`Cache::fromJson()` populates the hashes: https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v3.10.0/src/Cache/Cache.php#L129 .
so before we would decide to use or not use the Cache data from file due to version check, we already created Cache instance with `hashes` that are `array<string, int>` and not `array<string, string>`, as expected in your branch.

Or I'm missing sth?

Reply:
> ... https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v3.10.0/src/Cache/Cache.php#L129 . so before we would decide to use or not use the Cache data from file due to version check, we already created Cache instance with `hashes` that are `array<string, int>` and not `array<string, string>`, as expected in your branch.

the link is a clear proof the `hashes` from old/crc32 cache will be populated with `array<string, int>`

but my question is, is the cache used somewhere before the signature check? if not, why should be care about this usecase?

or you prefer the `hashes` in https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v3.10.0/src/Cache/Cache.php#L129 to be always casted to a string?

Reply:
if we claim, and not only with PHP Doc , but also with PHP (strict) type, that given value will be `string`, we should not populate it with anything else.
maybe _now_ we have a flow that given value is not in use, but it's very easy to see that code flow will change. And someone, without touching `get` function here, will start getting it reporting about wrong type on execution time (Eg https://3v4l.org/gje72 ). Let's avoid that.

Yeah, casting or validation can be a solution.

Reply:
casting fix landed